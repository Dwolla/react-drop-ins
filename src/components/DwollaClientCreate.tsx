import type { FC } from 'react';
import { ClientAccountOpeningAttributes } from '../@types/component-attributes';

const DwollaClientCreate: FC<ClientAccountOpeningAttributes<any>> = (props) => {
  return <dwolla-account-opening {...props} />;
};

export default DwollaClientCreate;
