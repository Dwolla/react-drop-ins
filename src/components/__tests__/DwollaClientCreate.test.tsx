import TestRenderer from 'react-test-renderer';
import DwollaClientCreate from '../DwollaClientCreate';

test('DwollaClientCreate', () => {
  const tree = TestRenderer.create(<DwollaClientCreate />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('with optional props', () => {
  const tree = TestRenderer.create(
    <DwollaClientCreate
      clientAccountId="123"
      correlationId="123abc"
      email="janedoe@email.com"
      firstName="Jane"
      lastName="Doe"
      ipAddress="255.255.255.255"
      hideDBAField
      privacy="www.yourprivacy.com"
      terms="www.yourterms.com"
      website="www.yourwebsite.com"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
