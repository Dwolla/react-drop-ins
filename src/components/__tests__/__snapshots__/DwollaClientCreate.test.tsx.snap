// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DwollaClientCreate 1`] = `<dwolla-account-opening />`;

exports[`with optional props 1`] = `
<dwolla-account-opening
  clientAccountId="123"
  correlationId="123abc"
  email="janedoe@email.com"
  firstName="Jane"
  hideDBAField={true}
  ipAddress="255.255.255.255"
  lastName="Doe"
  privacy="www.yourprivacy.com"
  terms="www.yourterms.com"
  website="www.yourwebsite.com"
/>
`;
